src/plugins/usage_collection/public/components/track_application_view.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Component, ReactNode } from 'react';
+import { ApplicationUsageTracker } from '@kbn/analytics';
+
+interface Props {
+  viewId: string;
+  applicationUsageTracker?: ApplicationUsageTracker;
```

Comment:
I wonder if the `usage_collection` plugin can own this instead of asking other plugins to send the API along to a component also exposed by `usage_collection`. What do you think?

Maybe this whole component can be implemented in `@kbn/analytics`, giving us more control about which clicks to count, etc?

Reply:
It is similar to how we pass the reporter to the telemetry services. It helps keep things decoupled without maintaining a state/singleton.

Regardless where things are implemented, we'd still have to pass the initialized tracker. For now the package contains all the API logic and barebones to get things running. And the plugin maintains the "moving" parts of the implementation. It helps keep the package code stable and flexible.

Reply:
> It is similar to how we pass the reporter to the telemetry services.

I think it's different in this case. The telemetry services are only used internally. The idea here is to provide an API that is as easy to use as possible. With the current implementation, devs have to spread the required prop all the way up to the parent components (as it happens in `src/plugins/telemetry_management_section/public/components/telemetry_management_section_wrapper.tsx`). I think this is less than ideal DX.

> It helps keep things decoupled without maintaining a state/singleton.

I understand, it just feels weird to me that a dev needs to import a component from a plugin, and send in the props an API that is provided by the same plugin (without applying any changes at all to it) 🤷 
Also, what would happen if we get multiple reporters initialised and started at the same time? Won't we have concurrency issues? Will they overwrite each other on the use of the local storage or any other side effects? Not defending the use of singletons everywhere, but maybe it's a good thing to not expose its state (and public methods of `ApplicationUsageTracker`) here. A dev could stop/break the whole Application Usage feature just by calling `applicationUsageTracker.stop()` at any given time.

Reply:
> What would happen if we get multiple reporters initialised and started at the same time? 

The plugin initializes 1 instance and that is assed to the components.

> A dev could stop/break the whole Application Usage feature just by calling applicationUsageTracker.stop() at any given time.

I can trim the start/stop functions from the start/setup contract to make sure we're not exposing devs calling `stop` accidentally :+1:.

> I understand, it just feels weird to me that a dev needs to import a component from a plugin, and send in the props an API that is provided by the same plugin (without applying any changes at all to it) 🤷

I dont see any way around this (and we do that almost everywhere, ie if you want a component to use the logger, you get the logger and pass it to the component, even if that componet is provided by core). Otherwise we'd have to maintain a singleton to store the tracker isntance state or pass react context. Both are not adviced practices.